scipy/optimize/_linprog_highs.py

```diff
@@ -232,14 +244,65 @@ def _linprog_highs(lp, solver, time_limit=None, presolve=True,
                 ``4`` : The HiGHS solver ran into a problem.
 
             nit : int
-                The total number of iterations performed in all phases.
+                The total number of iterations performed.
+                For ``solver='simplex'``, this includes iterations in all
+                phases. For ``solver='ipm'``, this does not include
+                crossover iterations.
             message : str
                 A string descriptor of the exit status of the algorithm.
 
     """
 
     _check_unknown_options(unknown_options)
 
+    warning_message = ("Option `dual_feasibility_tolerance` is {0}, "
```

Comment:
@AtsushiSakai per your request.

Reply:
@mdhaber  Probably best to match tolerances of previous linprog incarnations.  If there is no analog, then the HiGHS defaults seem reasonable to me.  As for `dual_objective_value_upper_bound`, is the checking that HiGHS does not enough?